deepchem/data/data_loader.py

```diff
@@ -968,42 +968,55 @@ def create_dataset(self,
     if isinstance(input_files, str):
       input_files = [input_files]
 
-    def shard_generator():  # TODO Enable sharding with shard size parameter
+    def shard_generator():
       for input_file in input_files:
-        if self.legacy:
+        shards = _read_file(input_file)
+        if self.legacy:  # Using legacy logic
           X = encode_bio_sequence(input_file)
-        else:
-          sequences = _read_file(input_file)
-          X = self.featurizer(sequences)
-        ids = np.ones(len(X))
-        # (X, y, w, ids)
-        yield X, None, None, ids
+          ids = np.ones(len(X))
+          # (X, y, w, ids)
+          yield X, None, None, ids
+        else:  # Not using legacy logic
+          for shard in shards:
+            X = self.featurizer(shard)
+            ids = np.ones(len(X))
+            # (X, y, w, ids)
+            yield X, None, None, ids
 
     def _read_file(input_file: str, auto_add_annotations: bool = False):
       """
       Convert the FASTA file to a numpy array of FASTA-format strings.
       """
 
-      # TODO don't convert all sequences into np array (allow shards)
```

Comment:
You'll want to changed this function to be shard size aware (you're probably already working on that!)

Reply:
I think this was addressed in our meeting!

The method body simply accesses the instance variable shard_size. Taking in an extra shard_size parameter might be a violation of the Liskov principle.